### PR TITLE
Use setuptools to build (get rid of deprecated distutils)

### PR DESCRIPTION
**Pull request by marcelotduarte**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -152,6 +149,6 @@ def run(self):
 
 # perform the setup
 setup(
-        cmdclass=dict(build_ext=build_ext, install_data=install_data),
+        cmdclass={"build_ext": build_ext, "install": install},
```
Comment (anthony-tuininga) [inline setup.py]: This change is spurious. Is there a reason why you want to get rid of the use of the dict() constructor?

File: setup.py [hunk 2]
```diff
@@ -50,40 +44,43 @@ def build_extension(self, ext):
...
+class install(setuptools.command.install.install):
 
     def run(self):
-        distutils.command.install_data.install_data.run(self)
+        super().run()
         if sys.platform == "win32":
+            # library (.lib or .a)
             command = self.get_finalized_command("build_ext")
-            target_dir = os.path.join(self.install_dir, "Libs")
+            if command.compiler.compiler_type == "msvc":
```
Comment (anthony-tuininga) [inline setup.py]: These changes have nothing to do with switching to setuptools (which I support). Technically they should be in a separate PR explaining their purpose.

File: setup.py (hunk 1 above)
Comment (marcelotduarte) [reply setup.py]: A new key (install) is added, so I changed to a common used form.

File: setup.py (hunk 2 above)
Comment (marcelotduarte) [reply setup.py]: They work differently in msvc and mingw.

File: setup.py (hunk 1 above)
Comment (anthony-tuininga) [reply setup.py]: Right. But why not just change the key? That's easier to follow and if you feel strongly enough about the formatting you could change it separately -- along with other things you might not like. ;-)

Comment (marcelotduarte) [reply setup.py]: Revert the change. I'd like to contribute a "black" format if you're not against it.

File: setup.py (hunk 2 above)
Comment (anthony-tuininga) [reply setup.py]: Sure. Sounds like that should be a separate PR with an explanation. Can you do that? I presume you have tested these changes on Windows yourself?

File: setup.py (hunk 1 above)
Comment (anthony-tuininga) [reply setup.py]: No objection. :-)

File: setup.py (hunk 2 above)
Comment (marcelotduarte) [reply setup.py]: I already reverted. I'll make a separate PR for mingw.

Comment (marcelotduarte) [reply setup.py]: Tested on windows and linux. This part on windows only.

Comment (anthony-tuininga) [reply setup.py]: Great. Thanks.
